src/modules/billing/services/two-part-tariff-service/index.js

```diff
@@ -0,0 +1,91 @@
+const { set } = require('lodash');
+const twoPartTariffMatching = require('./two-part-tariff-matching');
+const returnHelpers = require('./returns-helpers');
+const Purpose = require('../../../../lib/models/purpose');
+
+const mapResultsWithData = (overallError, data, transactions) => {
+  const updatedTransactions = [];
+  for (const result of data) {
+    const { error, data: { chargeElementId, actualReturnQuantity } } = result;
+    const [transaction] = transactions.filter(transaction => transaction.chargeElement.id === chargeElementId);
+
+    const twoPartTariffStatus = overallError || error;
+    set(transaction, 'calculatedVolume', actualReturnQuantity);
```

Comment:
The service models have a `fromHash` method for setting lots of properties, e.g.:
```
transaction.fromHash({
   volume : twoPartTariffStatus ? null : actualReturnQuantity,
   twoPartTariffStatus,
  ...
```